### ISSUE TITLE
Support for Visual Studio Code

### ISSUE THREAD

**Issue by pwab** [LiaScript/LiaScript#25]
[VS Code](https://code.visualstudio.com/) is a comparable editor to Atom and is used by alot of people. Maybe the reach of LiaScript could be extended by adding a liascript extension to its marketplace.

I'm not sure how the marketplace works but this could also benefit users of [vscodium](https://github.com/VSCodium/vscodium) or [code-server](https://github.com/cdr/code-server).

**Comment by andre-dietrich** [comment]
There is a new npm package, that I had published in order to simplify the development of editor extension/previews. It reflects the state of the [LiaScript/Editor branch](https://github.com/LiaScript/LiaScript/tree/editor). There is also some documentation about it

https://www.npmjs.com/package/@liascript/editor/v/0.8.12--1.0.0

The [preview-liascript](https://github.com/andre-dietrich/liascript-preview/) plugin for atom uses this package at the moment. The glue-code that is required can be found here:

https://github.com/andre-dietrich/liascript-preview/blob/master/lib/liascript-preview-view.js

In simple terms, actually it should be enough to open a webview or iframe and connect the editor with two functions, for double-click navigation/jumping and live-updates (if required). Thus, creating a plugin for an editor that supports JavaScript, should not be a big deal. If no one else does, I will try this a bit later :smiley:

**Comment by andre-dietrich** [comment]
It is not a VS-Code extension in general, since it seems to be quite tricky to create an extension, that dynamically loads JavaScript. However, there is now a dev-server available that can be installed locally, it monitors files and also updates the browser-view whenever there is a new update:

https://www.npmjs.com/package/@liascript/devserver

**Comment by andre-dietrich** [comment]
There are now two plugins available for vscode:

1. https://marketplace.visualstudio.com/items?itemName=LiaScript.liascript-preview
2. https://marketplace.visualstudio.com/items?itemName=LiaScript.liascript-preview-web
